code/modules/modular_computers/file_system/programs/system/file_browser.dm

```diff
@@ -73,8 +80,7 @@
 		if(!file || !istype(file))
 			return 1
 		var/newname = sanitize(input(usr, "Enter new file name:", "File rename", file.filename))
-		if(file && newname)
-			file.filename = newname
+		file.filename = newname
```

Comment:
Removing the two checks here is bad. First, we don't know if the name is empty; second, the file might get deleted while the proc sleeps. Put them back.